### PR TITLE
bug:51428 Use Azure Application Gateway instead of Azure Load Balancer in Azure Solution Template

### PR DESCRIPTION
**Pull request by Shell32-Natsu**

### CODE REVIEW

File: createUiDefinition.json [hunk 1]
```diff
@@ -203,9 +203,19 @@
...
                                     "defaultValue": {
-                                      "name": "vnet-subnet"
+                                      "name": "vnet-vm-subnet"
+                                    },
+                                    "constraints": {
+                                      "minAddressPrefixSize": "/24"
+                                    }
+                                  },
+                                  "subnet2": {
+                                    "label": "Applciation Gateway Subnet",
```
Comment (mattsunsjf) [inline createUiDefinition.json]: Can Application Gateway Subnet be in the same subnet of VMs?

File: mainTemplate.json [hunk 2]
```diff
@@ -150,7 +142,7 @@
             "minLength": 1,
             "defaultValue": "https://raw.githubusercontent.com/marklogic/cloud-enablement-azure/v9.0-6.1"
         },
-        "location": {
+        "Location": {
```
Comment (mattsunsjf) [inline mainTemplate.json]: Is this change necessary? Have you tried if it works with the createUiDefinition?

Comment (Shell32-Natsu) [reply mainTemplate.json]: The `Location` parameter in the createUiDefinition.json does have a capital *L*. I change this for this reason.
